### PR TITLE
Support substring matching in qube name text fields

### PR DESCRIPTION
**Pull request by alimirjamali**
fixes: https://github.com/QubesOS/qubes-issues/issues/8583

### CODE REVIEW

**Comment by alimirjamali** [comment]
Making Pylint happy(er)

**Review by marmarta — APPROVED** [review]

**Comment by marmarta** [comment]
PipelineRetryFailed

**Comment by alimirjamali** [comment]
Is this related to this commit? Or a policy error?

image

**Comment by marmarek** [comment]
The other qrexec PR in the batch is broken. Now rerunning tests without it.

File: qrexec/tools/qrexec_policy_agent.py [hunk 1]
```diff
@@ -184,11 +184,22 @@ def apply_model(
                 completion.set_inline_selection(True)
                 completion.set_inline_completion(True)
                 completion.set_popup_completion(True)
-                completion.set_popup_single_match(False)
+                completion.set_popup_single_match(True)
```
Comment (marmarek) [inline qrexec/tools/qrexec_policy_agent.py:187]: Can you explain this change? It means you can't accept a single match (after typing it in) with a single click anymore, you need two clicks. OpenQA found this out, but TBH I'm not sure if that's a common case.

Comment (alimirjamali) [reply qrexec/tools/qrexec_policy_agent.py:187]: > Can you explain this change?

That is something I found only after using the patch for a while. The default logic (matching from the start) does not require to show the popup for a single match if user has entered the entire string.

In our case, if user wants to search for `disp1234` and user has currenty `disp1234`, `dips5678` and `disp90ab` running at the moment and user types `1`, there will not popup result to select `disp1234` from it if `popup_single_match` is et to False.

Comment (alimirjamali) [reply qrexec/tools/qrexec_policy_agent.py:187]: p.s. Actually the GTK default matching algorithm is also insufficient if a popup is not shown. As it is case in-senstivie and Qube names are case-sensitive.

Comment (marmarek) [reply qrexec/tools/qrexec_policy_agent.py:187]: I see, ok.

**Pull request merged**